calico/common.py

```diff
@@ -96,14 +96,20 @@ def complete_logging(logfile=None,
 
     # If default_logging got called already, we'll have some loggers in place.
     # Update their levels.
-    for handler in root_logger.handlers:
+    for handler in root_logger.handlers[:]:
         if isinstance(handler, logging.handlers.SysLogHandler):
-            handler.setLevel(syslog_level)
+            if syslog_level is None:
+                root_logger.removeHandler(handler)
+            else:
+                handler.setLevel(syslog_level)
         elif isinstance(handler, logging.StreamHandler):
-            handler.setLevel(stream_level)
+            if stream_level is None:
+                root_logger.removeHandler(handler)
+            else:
+                handler.setLevel(stream_level)
 
     # If we've been given a log file, log to file as well.
-    if logfile is not None:
+    if logfile is not None and file_level is not None:
```

Comment:
I suggest broadening this check to `if not logfile and file_level is not None:`. The reason for doing this is that a user may accidentally put the following line in a logfile:

```python
LogFilePath = 
```

I don't *know* that this will lead to the `logfile` value being the empty string, but I'd certainly want to guard against the possibility.